### PR TITLE
Add signal filtering api

### LINKED ISSUE #158: Add signal filtering API
Provide configurable filters for each signal. Each filter interface would provide a method named `shouldInclude` that contains the final version of the signal's data object (before it gets sent to the exporter). Said function should return a boolean, where `true` would mean that the element will get sent to the backend, and `false` to filter it out.

### PR DESCRIPTION
**Pull request by LikeTheSalad**
Closes #158

### CODE REVIEW

File: android-sdk/src/main/java/co/elastic/apm/android/sdk/ElasticApmConfiguration.java [hunk 1]
```diff
@@ -122,6 +134,36 @@ public Builder setSignalConfiguration(SignalConfiguration signalConfiguration) {
             return this;
         }
 
+        /**
```
Comment (LikeTheSalad) [inline android-sdk/src/main/java/co/elastic/apm/android/sdk/ElasticApmConfiguration.java:137]: In the end, I figured that, at least in Android, setting the filters with a function named `set*` might be better in order to keep consistency with all the other config options @bryce-b @gregkalapos jsyk

**Comment by bryce-b** [comment]
I think we should allow for filter chaining. By this I mean allowing multiple filters to be set. Instead of `setFilter` we can have `addFilter`. I think this will allow customers a cleaner looking filter set up, if they want to have multiple checks.

**Comment by LikeTheSalad** [comment]
Sounds good, I just added filter chaining support @bryce-b

**Review by bryce-b — APPROVED** [review]

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -127,6 +127,45 @@ class MyApp extends android.app.Application {
 }
 ----
 
+[float]
+==== Signal filtering
+
+You can provide your own filters to decide what spans, logs and metrics are allowed to be exported to the backend.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:133]:
```suggestion
You can provide your own filters to specify which spans, logs, and metrics are allowed to be exported to the backend.
```

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -127,6 +127,45 @@ class MyApp extends android.app.Application {
...
+                    }
+                })
+                .build();
+        ElasticApmAgent.initialize(this, configuration);
+    }
+}
+----
+
+Each filter will contain a `shouldInclude` function which provides the signal item to be evaluated.
+This function must return a boolean value, `true` for when the provided item is allowed to continue, `false` when it must be discarded.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:165]:
```suggestion
This function must return a boolean value--`true` when the provided item is allowed to continue or `false` when it must be discarded.
```

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**